lib/stat_tracker.rb

```diff
@@ -3,49 +3,81 @@
 require_relative 'game_teams_factory'
 require_relative 'game_factory'
 
-
-# require './lib/team_factory'
-# require './lib/game_teams_factory'
-# require './lib/game_factory'
-
 class StatTracker 
-  
-  def initialize
-    @teams = []
-    @game_teams = []
-  end
 
   def from_csv(path)
     if path == './data/games.csv'
-      create_games_array(path)
+     @game_factory = create_games_factory(path)
     elsif path == './data/teams.csv'
-      create_teams_array(path)
+      @team_factory = create_teams_factory(path)
     else
-      './data/game_teams.csv'
-      create_game_teams_array(path)
+      @game_teams_factory = create_game_teams_factory(path)
     end
   end
 
-  def create_teams_array(path)
+  def create_teams_factory(path)
     team_factory = TeamFactory.new
     team_factory.create_teams(path)
-    #require 'pry'; binding.pry
+    team_factory
   end
   
-
-  def create_games_array(path)
+  def create_games_factory(path)
    game_factory = GameFactory.new
    game_factory.create_games(path)
-   #require 'pry'; binding.pry
+   game_factory
   end
 
-
-
-  def create_game_teams_array(path)
+  def create_game_teams_factory(path)
     game_teams_factory = GameTeamsFactory.new
     game_teams_factory.create_game_teams(path)
-    require 'pry'; binding.pry
+    game_teams_factory
+  end
+
+  def percentage_home_wins
+    home_wins = 0 
+    @game_factory.games.each do |game|
+      if game[:home_goals] > game[:away_goals]
+        home_wins += 1
+      end
+    end
+    percentage_wins = (home_wins.to_f / @game_factory.games.count.to_f)
+    percentage_wins.round(2)
+  end
+  
+  def percent_of_ties
+    ties = @game_factory.games.count do |game|
+      game[:away_goals] == game[:home_goals]
+    end
+  end
+
```

Comment:
this is were you added that percent_of_ties method back in but some how you did not in the full method.

Reply:
Is it just the one line  ` (ties.to_f / @games.length).round(2)` that this method needs? I don't see why it needs to be deleted out right and re-added with another branch. We can go in and add that line and then commit and this branch should work if I am understanding the problem correctly.